### DISCUSSION TITLE
Unable to include netbox_secrets in custom template

### DISCUSSION THREAD

**Discussion by jhofmueller** [netbox-community/netbox#14014] [Help Wanted!]
Hi!

I added the netbox_secrets plugin and try to render a device config including stored secrets. However I cannot find a way to access secrets from within a custom template. I read the [discussion](https://github.com/netbox-community/netbox/issues/12814) regarding objects exposed in the configuration rendering feature but either **plugins** are not included or I am making a mistake.

Any thoughts on that?

**Comment by DanSheps** [comment]
Are you referring to a configuration template?

**Reply by jhofmueller** [reply]
Yes.

**Reply by DanSheps** [reply]
I am not sure if plugins are included, however you can likely access the secrets by going through the device's reverse lookup (assuming there is one).  It looks like both device's and virtualmachines have it on their models via the secrets accessor (device.secrets.all()).

**Reply by jhofmueller** [reply]
Wonderful, that works! Thanks a lot! There is so much to learn :sunglasses:

**Reply by jhofmueller** [reply]
Part one - accessing the object - works. However now I am stuck at part two, decrypting the actual secret. I already posted the question at the [netbox-secrets forum](https://github.com/Onemind-Services-LLC/netbox-secrets/discussions/102) but so far nobody replied. Maybe anyone here has an idea?

**Reply by jhofmueller** [reply]
I removed the answered tag to get more attention here :sunglasses:
